content/en/blog/falcoctl-gitlab-oci-artifacts-support.md

```diff
@@ -0,0 +1,34 @@
+---
+title: "GitLab Container Registry Now Supports falcoctl OCI Artifact mediaTypes"
+linktitle: "GitLab Container Registry Now Supports falcoctl OCI Artifact mediaTypes"
+date: 2023-08-11
+author: Batuhan Apaydın
+slug: gitLab-container-registry-now-supports-falcoctl-oci-artifact-mediaTypes
+tags: ["Rules","Falco Plugins","Falcoctl", "GitLab", "OCI", "OCI Artifacts"]
+---
+
+Today, we are going to have a quick chit chat about the latest contribution we (w/[Emin Aktas](https://twitter.com/emminaktas)) made to GitLab's [Container Registry](https://gitlab.com/gitlab-org/container-registry).
+
+We noticed that GitLab Container Registry was not supporting falcoctl's OCI Artifact [mediaTypes]((https://github.com/falcosecurity/falcoctl/blob/7f1e8825a6f86010b9194577c56712dd0ef0442d/pkg/oci/constants.go#L20-L29) while we were pushing back the Falco's rules stored in GitHub's container registry to GitLab's container registry, so we decided to contribute to GitLab's Container Registry to add support for falcoctl's OCI Artifact mediaTypes.
+
+```shell
+Error: PUT https://registry.gitlab.com/v2/x/falcosecurity/rules/k8saudit-rules/manifests/1: MANIFEST_INVALID: manifest invalid; unknown media type: application/vnd.cncf.falco.rulesfile.config.v1+json
+Error: PUT https://registry.gitlab.com/v2/x/falcosecurity/plugins/k8saudit/manifests/sha256:b29c97a6590486f8b3b83644677e11d2f68e201a7035699189653d7f571d7e13: MANIFEST_INVALID: manifest invalid; unknown media type: application/vnd.cncf.falco.plugin.config.v1+json
+```
+
+Then, we have added support for falcoctl's OCI Artifact [mediaTypes](https://github.com/falcosecurity/falcoctl/blob/7f1e8825a6f86010b9194577c56712dd0ef0442d/pkg/oci/constants.go#L20-L29) to GitLab Container Registry, you can learn more about the contribution, [here](https://gitlab.com/gitlab-org/container-registry/-/merge_requests/1375) which means that you can now be able to pull and push falcoctl OCI Artifacts from and to GitLab Container Registry once the feature is released, which is planned to be released with GitLab **16.3**.
```

Comment:
```suggestion
You can learn more about our contribution [here](https://gitlab.com/gitlab-org/container-registry/-/merge_requests/1375). It will allow you to pull and push falcoctl OCI Artifacts from and to GitLab Container Registry once the feature is released, planned for GitLab **16.3**.
```

Reply:
less is more okay, I'll accept that, thanks.